internal/aks/client.go

```diff
@@ -0,0 +1,152 @@
+	}
+
+	client := resources.NewGroupsClientWithBaseURI(to.String(baseURL), subscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+func NewClusterClient(cred *Credentials) (*containerservice.ManagedClustersClient, error) {
+	authorizer, err := NewClientAuthorizer(cred)
+	if err != nil {
+		return nil, err
+	}
+
+	baseURL := cred.BaseURL
+	if baseURL == nil {
+		baseURL = to.StringPtr(azure.PublicCloud.ResourceManagerEndpoint)
+	}
+
+	subscriptionID := cred.SubscriptionID
+	if subscriptionID == "" {
+		return nil, fmt.Errorf("subscriptionId not specified")
+	}
+
+	client := containerservice.NewManagedClustersClientWithBaseURI(to.String(baseURL), subscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+func NewAgentPoolClient(cred *Credentials) (*containerservice.AgentPoolsClient, error) {
+	authorizer, err := NewClientAuthorizer(cred)
+	if err != nil {
+		return nil, err
+	}
+
+	baseURL := cred.BaseURL
+	if baseURL == nil {
+		baseURL = to.StringPtr(azure.PublicCloud.ResourceManagerEndpoint)
+	}
+
+	subscriptionID := cred.SubscriptionID
+	if subscriptionID == "" {
+		return nil, fmt.Errorf("subscriptionId not specified")
+	}
+
+	agentProfile := containerservice.NewAgentPoolsClientWithBaseURI(to.String(baseURL), subscriptionID)
+	agentProfile.Authorizer = authorizer
+
+	return &agentProfile, nil
+}
+
+func NewClientAuthorizer(cred *Credentials) (autorest.Authorizer, error) {
+	authBaseURL := cred.AuthBaseURL
+	if authBaseURL == nil {
+		authBaseURL = to.StringPtr(azure.PublicCloud.ActiveDirectoryEndpoint)
+	}
+	baseURL := cred.BaseURL
+	if baseURL == nil {
+		baseURL = to.StringPtr(azure.PublicCloud.ResourceManagerEndpoint)
+	}
+
+	oauthConfig, err := adal.NewOAuthConfig(to.String(authBaseURL), cred.TenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	spToken, err := adal.NewServicePrincipalToken(*oauthConfig, cred.ClientID, cred.ClientSecret, to.String(baseURL))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't authenticate to Azure cloud with error: %v", err)
+	}
+
+	return autorest.NewBearerAuthorizer(spToken), nil
+}
+
+func GetSecrets(secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfigSpec) (*Credentials, error) {
+	var cred Credentials
+
+	secretName := spec.AzureCredentialSecret
+	if secretName == "" {
+		return nil, fmt.Errorf("secret name not provided")
+	}
+
+	ns, id := utils.ParseSecretName(secretName)
+	secret, err := secretsCache.Get(ns, id)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't find secret [%s] in namespace [%s]", id, ns)
+	}
+
+	clientIDBytes := secret.Data["azurecredentialConfig-clientId"]
+	clientSecretBytes := secret.Data["azurecredentialConfig-clientSecret"]
+	if clientIDBytes == nil || clientSecretBytes == nil {
+		return nil, fmt.Errorf("invalid secret client data for Azure cloud")
+	}
+	cred.ClientID = string(clientIDBytes)
+	cred.ClientSecret = string(clientSecretBytes)
+
+	if spec.AuthBaseURL != nil {
```

Comment:
I don't think you need to check for nil or `""` for these, you can just assign directly.

Reply:
Fixed